### PR TITLE
Hoverservice has it's own contextview instance

### PR DESCRIPTION
**Pull request by benibenj**
This allows features such as breadcrumbs to work

### CODE REVIEW

File: src/vs/platform/contextview/browser/contextViewService.ts [hunk 1]
```diff
@@ -3,18 +3,17 @@
...
+import { ContextView, ContextViewDOMPosition, IContextViewProvider } from 'vs/base/browser/ui/contextview/contextview';
 import { Disposable, IDisposable, toDisposable } from 'vs/base/common/lifecycle';
 import { ILayoutService } from 'vs/platform/layout/browser/layoutService';
 import { IContextViewDelegate, IContextViewService } from './contextView';
 import { getWindow } from 'vs/base/browser/dom';
 
-export class ContextViewService extends Disposable implements IContextViewService {
 
-	declare readonly _serviceBrand: undefined;
+export class ContextViewHandler extends Disposable implements IContextViewProvider {
```
Comment (bpasero) [inline src/vs/platform/contextview/browser/contextViewService.ts:13]: Is there a way to remove the DOM elements when they are hidden? I think now we pile up context view DOM elements on the root for each instance.

**Review by bpasero — CHANGES_REQUESTED** [review]
On a high level, can we now enable custom hovers for controls that are inside a context view, such as the zoom control?

**Comment by Eude1** [comment]
não estou antenter o que disseste

A dom, 25/02/2024, 07:43, Benjamin Pasero ***@***.***>
escreveu:

File: src/vs/platform/contextview/browser/contextViewService.ts (hunk 1 above)
Comment (benibenj) [reply src/vs/platform/contextview/browser/contextViewService.ts:13]: Not sure this is worth it for hover views as they can appear / reappear a lot. But makes sense for context views

**Comment by benibenj** [comment]
@bpasero 
> On a high level, can we now enable custom hovers for controls that are inside a context view, such as the zoom control?

Yes and no. It will work if you open a context view (using context view service) and then have a hover in that context view which uses the hover service. For example breadcrumbs context view. The Zoom control is a hover and so this would not work.

I had a quick go at implementing a hover stack, supporting hovers in hovers in hovers ... However, this is rather complicated to get working well with the current implementation of hovers and so would take a while to properly design if needed.

**Review by bpasero — APPROVED** [review]

**Pull request merged**